app/utils/verified-builds.tsx

```diff
@@ -39,111 +55,171 @@ export function useVerifiedProgramRegistry({
-            if (!pdaAccountInfo || !pdaAccountInfo.data) {
-                console.log('PDA account info not found');
+
+            const pdaAccountInfo = await (accountAnchorProgram.account as any).buildParams.fetch(pda);
+            if (!pdaAccountInfo) {
                 return null;
             }
-            return accountAnchorProgram?.coder.accounts.decode('buildParams', pdaAccountInfo.data);
+            return pdaAccountInfo;
         },
         { suspense: options?.suspense }
     );
 
-    const isLoading = isRegistryLoading || isPdaLoading;
-
-    if (registryError || pdaError) {
-        return { data: null, error: registryError || pdaError, isLoading };
+    if (!osecInfo || pdaError) {
+        return { data: null, error: pdaError, isLoading: isPdaLoading };
     }
-
-    // Create command from the args of the verify PDA
-    if (registryData && pdaData && !isLoading) {
-        const verifiedData = registryData as OsecRegistryInfo;
-        verifiedData.verify_command = `solana-verify verify-from-repo -um --program-id ${programId.toBase58()} ${
-            pdaData.gitUrl
-        }`;
-
-        if (pdaData.commit) {
-            verifiedData.verify_command += ` --commit-hash ${pdaData.commit}`;
-        }
-
-        // Add additional args if available, for example mount-path and --library-name
-        if (pdaData.args && pdaData.args.length > 0) {
-            const filteredArgs = [];
-
-            for (let i = 0; i < pdaData.args.length; i++) {
-                const arg = pdaData.args[i];
-
-                if (arg === '-b' || arg === '--base-image') {
-                    i++; // Also skip the parameter
-                    continue;
-                }
-                filteredArgs.push(arg);
-            }
-
-            if (filteredArgs.length > 0) {
-                const argsString = filteredArgs.join(' ');
-                verifiedData.verify_command += ` ${argsString}`;
-            }
-        }
-        verifiedData.repo_url = pdaData.gitUrl;
-        if (registryData.verification_status === VerificationStatus.NotVerified) {
-            verifiedData.message = 'Verify command was provided by the program authority.';
-            verifiedData.verification_status = VerificationStatus.PdaUploaded;
-        }
-        return { data: verifiedData, isLoading };
+    if (isPdaLoading) {
+        return { data: null, isLoading: isPdaLoading };
     }
-    if (registryData && pdaData == null && !isLoading) {
-        const verifiedData = registryData as OsecRegistryInfo;
 
-        verifiedData.verify_command = isMainnet(cluster)
+    const message = TRUSTED_SIGNERS[osecInfo?.signer || '']
+        ? 'Verification information provided by a trusted signer.'
+        : 'Verification information provided by the program authority.';
+
+    const enrichedOsecInfo: OsecRegistryInfo = {
+        ...osecInfo,
+        message,
+        signer: osecInfo?.signer || '',
+        verification_status: VerificationStatus.Verified,
```

Comment:
Is this always VerificationStatus.Verified or could it also not be verified?  I think this would need to come from the osecInfo?

Reply:
Great catch! Patched!